From f3e229ce724751d9ceb7cd1ecb86c42ec39ff231 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Thu, 10 Oct 2019 22:50:01 +0200
Subject: [PATCH] Do not mangle kill switch naming

---
 source/libnormaliz/nmz_nauty.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/libnormaliz/nmz_nauty.cpp b/source/libnormaliz/nmz_nauty.cpp
index 0503614..83196c5 100644
--- a/source/libnormaliz/nmz_nauty.cpp
+++ b/source/libnormaliz/nmz_nauty.cpp
@@ -37,6 +37,10 @@
 // #define MAXN 5000    /* Define this before including nauty.h */
 // we use dynamic allocation
 
+extern "C" {
+  extern volatile int nauty_kill_request;
+}
+
 #include <nauty/nauty.h>
 
 namespace libnormaliz {
-- 
2.23.0

